core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/WorkspaceMigration.scala

```diff
@@ -54,145 +56,151 @@ final case class WorkspaceMigration(id: Long,
+
+    final def isMigrating(workspace: Workspace): ReadAction[Boolean] =
+      sql"select count(*) from #$tableName where #$workspaceId = ${workspace.workspaceIdAsUUID} and #$started is not null and #$finished is null".as[Int].map(_.head > 0)
 
-  val workspaceMigrations = TableQuery[WorkspaceMigrationHistory]
+    final def schedule(workspace: Workspace, unlockOnCompletionBool: Boolean): WriteAction[Int] = {
+      sqlu"insert into #$tableName (#$workspaceId, #$unlockOnCompletion) values (${workspace.workspaceIdAsUUID}, $unlockOnCompletionBool)"
+    }
+
+    final def getMigrationAttempts(workspace: Workspace): ReadAction[List[WorkspaceMigration]] =
+      sql"select #$allColumns from #$tableName where #$workspaceId = ${workspace.workspaceIdAsUUID} order by #$id".as[WorkspaceMigration].map(_.toList)
+
+    final def selectMigrations(conditions: SQLActionBuilder): ReadAction[Vector[WorkspaceMigration]] = {
+      val startingSql = sql"select #$allColumns from #$tableName where #$finished is null and "
+      concatSqlActions(startingSql, conditions, sql" order by #$updated asc limit 1").as[WorkspaceMigration]
+    }
+
+    final def getAttempt(workspaceUuid: UUID): ReadAction[Option[WorkspaceMigration]] =
+      sql"select #$allColumns from #$tableName where #$workspaceId = $workspaceUuid order by #$id desc limit 1".as[WorkspaceMigration].headOption
+
+    final def getAttempt(id: Long): ReadAction[Option[WorkspaceMigration]] =
+      sql"select #$allColumns from #$tableName where #$id = $id".as[WorkspaceMigration].headOption
+
+    final def truncate: WriteAction[Int] = sqlu"delete from #$tableName"
+
+    val startCondition = sql"#$started is null"
+    val removeWorkspaceBucketIamCondition = sql"#$started is not null and #$workspaceBucketIamRemoved is null"
+    val claimAndConfigureGoogleProjectCondition = sql"#$workspaceBucketIamRemoved is not null and #$newGoogleProjectConfigured is null"
+    val createTempBucketConditionCondition = sql"#$newGoogleProjectConfigured is not null and #$tmpBucketCreated is null"
+    val issueTransferJobToTmpBucketCondition = sql"#$tmpBucketCreated is not null and #$workspaceBucketTransferJobIssued is null"
+    val deleteWorkspaceBucketCondition = sql"#$workspaceBucketTransferred is not null and #$workspaceBucketDeleted is null"
+    val createFinalWorkspaceBucketCondition = sql"#$workspaceBucketDeleted is not null and #$finalBucketCreated is null"
+    val issueTransferJobToFinalWorkspaceBucketCondition = sql"#$finalBucketCreated is not null and #$tmpBucketTransferJobIssued is null"
+    val deleteTemporaryBucketCondition = sql"#$tmpBucketTransferred is not null and #$tmpBucketDeleted is null"
+    val restoreIamPoliciesAndUpdateWorkspaceRecordCondition = sql"#$tmpBucketDeleted is not null"
+
+    def transferJobSucceededCondition(migrationId: Long) = sql"#$id = $migrationId"
```

Comment:
you need to read the transfer job table for the "Success" outcome

Reply:
Maybe this is named poorly. The name comes from the function in the actor that calls it

Reply:
Ah . Yeah this `condiition` stuff is a bit opaque. Maybe name it to `withMigrationId` so it's clearer?

Reply:
Or just inline the `conditions`?

Reply:
they can't be inlined because the `sql` interpolator needs the driver to be imported